Libraries/CommonFunctions.psm1

```diff
@@ -1956,61 +1956,72 @@ function Enable-RootUser {
-                }
+    if ($CompatibleDistro) {
+        if ($CompatibleDistro.Contains($DetectedDistro)) {
+            Write-LogInfo "Confirmed supported distro: $DetectedDistro"
+            return $true
+        } else {
+            Write-LogWarn "Unsupported distro: $DetectedDistro"
+            return $false
         }
-        for($i=0; $i -lt $actualKernelVersions.Length;$i++) {
-            try {
-                    $actualKernelVersions[$i] = [int]$actualKernelVersions[$i]
-                } catch {
-                    $actualKernelVersions[$i] = 0
-                    continue
-                }
+    } else {
+        # Supported Distro and kernel version for DPDK on Azure
+        # https://docs.microsoft.com/en-us/azure/virtual-network/setup-dpdk
+        $SUPPORTED_DISTRO_KERNEL = @{
+            "UBUNTU" = "4.15.0-1015-azure";
+            "SLES" = "4.12.14-5.5-azure";
+            "SLES 15" = "4.12.14-5.5-azure";
+            "SUSE" = "4.12.14-5.5-azure";
+            "REDHAT" = "3.10.0-862.9.1.el7";
+            "CENTOS" = "3.10.0-862.3.3.el7";
         }
 
-        $array_count = $actualKernelVersions.Length
-        if ($supportKernelVersions.Length -gt $actualKernelVersions.Length) {
-            $array_count = $supportKernelVersions.Length
-        }
+        if ($SUPPORTED_DISTRO_KERNEL.Keys -contains $DetectedDistro) {
+            $supportKernelVersions = $SUPPORTED_DISTRO_KERNEL[$DetectedDistro] -split "[\.\-]+"
+            $KernelVersions = $KernelVersion -split "[\.\-]+"
+            for($i=0; $i -lt $supportKernelVersions.Length;$i++) {
+                try {
+                        $supportKernelVersions[$i] = [int]$supportKernelVersions[$i]
+                    } catch {
+                        $supportKernelVersions[$i] = 0
+                        continue
+                    }
+            }
+            for($i=0; $i -lt $KernelVersions.Length;$i++) {
+                try {
+                        $KernelVersions[$i] = [int]$KernelVersions[$i]
+                    } catch {
+                        $KernelVersions[$i] = 0
+                        continue
+                    }
+            }
 
-        for($i=0; $i -lt $array_count;$i++) {
-            if ([int]$actualKernelVersions[$i] -eq [int]$supportKernelVersions[$i]) {
-                continue
-            } elseif ([int]$actualKernelVersions[$i] -lt [int]$supportKernelVersions[$i]) {
-                return $false
-            } else {
-                return $true
+            $array_count = $KernelVersions.Length
+            if ($supportKernelVersions.Length -gt $KernelVersions.Length) {
+                $array_count = $supportKernelVersions.Length
+            }
+
+            for($i=0; $i -lt $array_count;$i++) {
+                if ([int]$KernelVersions[$i] -eq [int]$supportKernelVersions[$i]) {
+                    continue
+                } elseif ([int]$KernelVersions[$i] -lt [int]$supportKernelVersions[$i]) {
```

Comment:
Add logging for unsupported kernel?

Reply:
Log in the invoke ps script, like https://github.com/LIS/LISAv2/pull/223/files#diff-2e65f1f3c5fd894b9e501ecfee206f3bR155

Reply:
> Add logging for unsupported kernel?

Log in the invoke ps script, like https://github.com/LIS/LISAv2/pull/223/files#diff-2e65f1f3c5fd894b9e501ecfee206f3bR155